src/components/Auth/TodoRender/TodoRender.tsx

```diff
@@ -0,0 +1,194 @@
+import classNames from 'classnames';
+import {
+  KeyboardEvent, useContext, useEffect, useRef,
+} from 'react';
+import { deleteTodo, updateTodo } from '../../../api/todos';
+import { TypeChange } from '../../../types/TypeChange';
+import { Todo } from '../../../types/Todo';
+import { TodoContext } from '../../../context/TodoContext';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoRender: React.FC<Props> = ({
+  todo,
+}) => {
+  const {
+    handleStatusChange,
+    selectedTodoId,
+    setSelectedTodoId,
+    setInputValue,
+    inputValue,
+    handleChangeTitle,
+    setLoadError,
+    setErrorMessage,
+    allCompletedLoader,
+    todoIdLoader,
+    setTodoIdLoader,
+    toggleLoader,
+  } = useContext(TodoContext);
+
+  const {
+    title, completed, id,
+  } = todo;
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const removeFromServer = async (data: Todo) => {
+    try {
+      setTodoIdLoader(data.id);
+      await deleteTodo(data.id);
+      handleStatusChange(data, TypeChange.delete);
+    } catch (_) {
+      setLoadError(true);
+      setErrorMessage('Unable to delete todo from the server');
+    } finally {
+      setTodoIdLoader(null);
+      setSelectedTodoId(-1);
+    }
+  };
+
+  const handleRemoveTodo = (data: Todo) => {
+    removeFromServer(data);
+  };
+
+  const changeCompleteOnServer = async () => {
```

Comment:
you can create one method for updating the todo that receive the todo id and data what need to update for this todo. And this method send request to server, server return updated todo and you can use this updated todo to update your local state

Reply:
I created method for updating 2 different todos but I'm not sure If I fixed what I needed to fix